src/include/duckdb/common/virtual_file_system.hpp

```diff
@@ -82,8 +83,10 @@ class VirtualFileSystem : public FileSystem {
 	}
 
 private:
+	FileSystem &FindFileSystem(const string &path, optional_ptr<FileOpener> file_opener);
+	FileSystem &FindFileSystem(const string &path, optional_ptr<ClientContext> client_context);
 	FileSystem &FindFileSystem(const string &path);
-	FileSystem &FindFileSystemInternal(const string &path);
+	FileSystem &FindFileSystemInternal(const string &path, bool &allow_autoloading_file_system);
```

Comment:
I'm not a huge fan of the bool reference output parameter. I think we could simplify this by making 
`FindFileSystemInternal` return an optional_ptr to a filesystem that is checked on level up:

```C++
auto fs_lookup = FindFileSystemInternal(path);
if (!fs_lookup && can_autoload) {
    Autoload();
    fs_lookup = FindFileSystemInternal(path);
}

if (fs_lookup) {
    return fs_lookup;
} else {
    return *default_fs;
}